### ISSUE TITLE
clojure.spec :fn equivalent?

### ISSUE THREAD

**Issue by knubie** [metosin/malli#764]
In clojure.spec you can do something like this:

```clj
(s/fdef ranged-rand
  :args (s/and (s/cat :start int? :end int?)
               #(< (:start %) (:end %)))
  :ret int?
  :fn (s/and #(>= (:ret %) (-> % :args :start))
             #(< (:ret %) (-> % :args :end))))
```

Where the `:fn` can be used to verify the relationship between args and return value.

Is there any way to do this in malli?

**Comment by ikitommi** [comment]
No, there is not yet such a thing, but it would be easy to add. Interested in doing a PR for that?

**Comment by knubie** [comment]
Well, if it's easy, then yeah I can give it a shot. 😎 I'll admit I haven't looked at the source code. Can you give me any pointers on where to start?

**Comment by ikitommi** [comment]
You could start from `malli.core/-instrument`. It's the (pure) functional core behind the functions schemas, everything else builds on top of that.

**Comment by ikitommi** [comment]
https://github.com/metosin/malli/blob/master/docs/function-schemas.md#instrumentation

**Comment by ikitommi** [comment]
fixed in #1000
